<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Chatbot</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2" columnstretch="3,1">
    <item row="1" column="0">
     <layout class="QGridLayout" name="centralGridLayout">
      <item row="0" column="0">
       <widget class="QTabWidget" name="mainTabWidget">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="trainTab">
         <attribute name="title">
          <string>Entrenar</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="2">
           <layout class="QGridLayout" name="trainTabGridLayout" columnstretch="2,0,0,3">
            <item row="1" column="0" rowspan="5" colspan="2">
             <widget class="QTreeWidget" name="categoriesTree">
              <property name="mouseTracking">
               <bool>true</bool>
              </property>
              <column>
               <property name="text">
                <string>Reglas</string>
               </property>
              </column>
              <item>
               <property name="text">
                <string>Saludos</string>
               </property>
               <item>
                <property name="text">
                 <string>Hola</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Buen día</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Buenas tardes</string>
                </property>
               </item>
              </item>
              <item>
               <property name="text">
                <string>Preguntas Personales</string>
               </property>
               <item>
                <property name="text">
                 <string>¿Cuál es tu nombre?</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>¿Cuantos años tienes?</string>
                </property>
               </item>
              </item>
             </widget>
            </item>
            <item row="0" column="2" rowspan="6">
             <spacer name="trainTabSpacer1">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType">
               <enum>QSizePolicy::Fixed</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>10</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="0" column="0" colspan="2">
             <layout class="QHBoxLayout" name="rulesToolbarLayout">
              <item>
               <widget class="QToolButton" name="addRuleButton">
                <property name="toolTip">
                 <string>Agregar regla</string>
                </property>
                <property name="text">
                 <string>+R</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="rmRuleButton">
                <property name="toolTip">
                 <string>Borrar regla</string>
                </property>
                <property name="text">
                 <string> - R</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="addCategoryButton">
                <property name="toolTip">
                 <string>Agregar categoría</string>
                </property>
                <property name="text">
                 <string>+C</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="rmCategoryButton">
                <property name="toolTip">
                 <string>Borrar categoría</string>
                </property>
                <property name="text">
                 <string>-C</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="rulesToolbarSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </item>
            <item row="0" column="3">
             <widget class="QLabel" name="ifUserWritesLabel">
              <property name="text">
               <string>Si el usuario escribe:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="3">
             <widget class="QLabel" name="chatbotRepliesLabel">
              <property name="text">
               <string>El chatbot responde:</string>
              </property>
             </widget>
            </item>
            <item row="3" column="3" rowspan="3">
             <widget class="QPlainTextEdit" name="ruleOutputText">
              <property name="mouseTracking">
               <bool>true</bool>
              </property>
              <property name="plainText">
               <string>Mi nombre es MiBot</string>
              </property>
             </widget>
            </item>
            <item row="1" column="3">
             <widget class="QPlainTextEdit" name="ruleInputText">
              <property name="plainText">
               <string>Cual es tu nombre?
Como te llamas?</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="chatTab">
         <attribute name="title">
          <string>Conectar con chat</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_8">
          <item row="0" column="1">
           <layout class="QFormLayout" name="chatTabFormLayout">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
            </property>
            <item row="0" column="1">
             <spacer name="chatTabSpacer1">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="connectWithChatLabel">
              <property name="font">
               <font>
                <weight>50</weight>
                <bold>false</bold>
               </font>
              </property>
              <property name="text">
               <string>Conectar con chat:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QRadioButton" name="fbChatRadio">
              <property name="text">
               <string>Facebook</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QRadioButton" name="gtalkChatRadio">
              <property name="text">
               <string>Gmail o Google Talk</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <spacer name="chatTabSpacer2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="usernameLabel">
              <property name="text">
               <string>Usuario:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QLineEdit" name="usernameText">
              <property name="maximumSize">
               <size>
                <width>150</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>andres.pagliano</string>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QLabel" name="passwordLabel">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Contraseña:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="5" column="1">
             <widget class="QLineEdit" name="passwordText">
              <property name="maximumSize">
               <size>
                <width>150</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>xxxxx</string>
              </property>
              <property name="echoMode">
               <enum>QLineEdit::Password</enum>
              </property>
             </widget>
            </item>
            <item row="6" column="1">
             <widget class="QPushButton" name="connectButton">
              <property name="maximumSize">
               <size>
                <width>100</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="text">
               <string>Conectar</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="conversationsTab">
         <attribute name="title">
          <string>Ver conversaciones</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_9">
          <item row="0" column="0">
           <layout class="QGridLayout" name="conversationTabGridLayout" columnstretch="2,3">
            <item row="1" column="0">
             <widget class="QTableWidget" name="conversationContactsTable">
              <property name="font">
               <font>
                <pointsize>8</pointsize>
               </font>
              </property>
              <property name="selectionMode">
               <enum>QAbstractItemView::SingleSelection</enum>
              </property>
              <property name="selectionBehavior">
               <enum>QAbstractItemView::SelectRows</enum>
              </property>
              <attribute name="horizontalHeaderStretchLastSection">
               <bool>true</bool>
              </attribute>
              <attribute name="verticalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <row>
               <property name="text">
                <string>New Row</string>
               </property>
              </row>
              <row>
               <property name="text">
                <string>New Row</string>
               </property>
              </row>
              <column>
               <property name="text">
                <string>Fecha/Hora</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Contacto</string>
               </property>
              </column>
              <item row="0" column="0">
               <property name="text">
                <string>10/02/12 12:22</string>
               </property>
              </item>
              <item row="0" column="1">
               <property name="text">
                <string>paglian@gmail.com</string>
               </property>
              </item>
              <item row="1" column="0">
               <property name="text">
                <string>09/02/12 16:55</string>
               </property>
              </item>
              <item row="1" column="1">
               <property name="text">
                <string>pablo@pablopda.com.ar</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="0" column="0" colspan="2">
             <widget class="QLabel" name="conversationHeaderLabel">
              <property name="font">
               <font>
                <weight>75</weight>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>Conversaciones mantenidas con el chatbot:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QTableWidget" name="conversationTable">
              <property name="font">
               <font>
                <pointsize>8</pointsize>
               </font>
              </property>
              <property name="horizontalScrollBarPolicy">
               <enum>Qt::ScrollBarAlwaysOff</enum>
              </property>
              <property name="editTriggers">
               <set>QAbstractItemView::NoEditTriggers</set>
              </property>
              <property name="alternatingRowColors">
               <bool>true</bool>
              </property>
              <property name="selectionMode">
               <enum>QAbstractItemView::NoSelection</enum>
              </property>
              <attribute name="horizontalHeaderStretchLastSection">
               <bool>true</bool>
              </attribute>
              <attribute name="verticalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <row>
               <property name="text">
                <string>Contacto</string>
               </property>
              </row>
              <row>
               <property name="text">
                <string>Chatbot</string>
               </property>
              </row>
              <row>
               <property name="text">
                <string>Contacto</string>
               </property>
              </row>
              <row>
               <property name="text">
                <string>Chatbot</string>
               </property>
              </row>
              <row>
               <property name="text">
                <string>Contacto</string>
               </property>
              </row>
              <row>
               <property name="text">
                <string>Chatbot</string>
               </property>
              </row>
              <column>
               <property name="text">
                <string>Hora</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Conversación</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>Estado</string>
               </property>
              </column>
              <item row="0" column="0">
               <property name="text">
                <string>10:23:12</string>
               </property>
              </item>
              <item row="0" column="1">
               <property name="text">
                <string>Hola</string>
               </property>
              </item>
              <item row="0" column="2">
               <property name="text">
                <string/>
               </property>
              </item>
              <item row="1" column="0">
               <property name="text">
                <string>10:23:14</string>
               </property>
              </item>
              <item row="1" column="1">
               <property name="text">
                <string>Hola!</string>
               </property>
              </item>
              <item row="1" column="2">
               <property name="text">
                <string>Ok</string>
               </property>
               <property name="textAlignment">
                <set>AlignHCenter|AlignVCenter|AlignCenter</set>
               </property>
               <property name="foreground">
                <brush brushstyle="NoBrush">
                 <color alpha="255">
                  <red>85</red>
                  <green>170</green>
                  <blue>0</blue>
                 </color>
                </brush>
               </property>
              </item>
              <item row="2" column="0">
               <property name="text">
                <string>10:23:19</string>
               </property>
              </item>
              <item row="2" column="1">
               <property name="text">
                <string>Como te llamas?</string>
               </property>
              </item>
              <item row="3" column="0">
               <property name="text">
                <string>10:23:20</string>
               </property>
              </item>
              <item row="3" column="1">
               <property name="text">
                <string>MiBot</string>
               </property>
              </item>
              <item row="3" column="2">
               <property name="text">
                <string>Ok</string>
               </property>
               <property name="textAlignment">
                <set>AlignHCenter|AlignVCenter|AlignCenter</set>
               </property>
               <property name="foreground">
                <brush brushstyle="NoBrush">
                 <color alpha="255">
                  <red>85</red>
                  <green>170</green>
                  <blue>0</blue>
                 </color>
                </brush>
               </property>
              </item>
              <item row="4" column="0">
               <property name="text">
                <string>10:23:23</string>
               </property>
              </item>
              <item row="4" column="1">
               <property name="text">
                <string>Cuanto es 2+2?</string>
               </property>
              </item>
              <item row="5" column="0">
               <property name="text">
                <string>10:23:27</string>
               </property>
              </item>
              <item row="5" column="1">
               <property name="text">
                <string>No entiendo tu pregunta</string>
               </property>
              </item>
              <item row="5" column="2">
               <property name="text">
                <string>Falló</string>
               </property>
               <property name="textAlignment">
                <set>AlignHCenter|AlignVCenter|AlignCenter</set>
               </property>
               <property name="background">
                <brush brushstyle="NoBrush">
                 <color alpha="255">
                  <red>0</red>
                  <green>0</green>
                  <blue>0</blue>
                 </color>
                </brush>
               </property>
               <property name="foreground">
                <brush brushstyle="SolidPattern">
                 <color alpha="255">
                  <red>255</red>
                  <green>0</green>
                  <blue>0</blue>
                 </color>
                </brush>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="1">
     <widget class="QTabWidget" name="rightSideTabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="helpTab">
       <attribute name="title">
        <string>Ayuda</string>
       </attribute>
       <widget class="QLabel" name="helpLabel">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>221</width>
          <height>221</height>
         </rect>
        </property>
        <property name="text">
         <string>La ayuda se actualiza en esta zona a medida que el usuario mueve el mouse sobre los controles</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="testTab">
       <attribute name="title">
        <string>Prueba tu chatbot</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QLabel" name="testLabel">
          <property name="text">
           <string>Prueba:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="testConversationText">
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="html">
           <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Cantarell'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#0000ff;&quot;&gt; - Hola&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#550000;&quot;&gt;Hola!&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#0000ff;&quot;&gt; - ¿Como te llamas?&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; color:#550000;&quot;&gt;Mi nombre es MiBot&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="testInputText">
          <property name="text">
           <string>Cuantos años tie</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <action name="actionAbrir">
   <property name="text">
    <string>Abrir...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionGuardar">
   <property name="text">
    <string>Guardar</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSalir">
   <property name="text">
    <string>Salir</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionGuardar_como">
   <property name="text">
    <string>Guardar como...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionMostrar_ayuda">
   <property name="text">
    <string>Mostrar ayuda</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionReportar_un_error">
   <property name="text">
    <string>Reportar un error</string>
   </property>
  </action>
  <action name="actionAcerca_de">
   <property name="text">
    <string>Acerca de...</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
